### PR TITLE
ACSD 44733: Cancelled Orders Are Not Changing Inventory Properly

### PR DESCRIPTION
**Pull request by saddane**
This is patch includes: MDVA-37291 and ACP2E-924
Combining both patched to avoid conflicts.

### CODE REVIEW

File: InventoryReservationCli/Model/SalableQuantityInconsistency/LoadExistingReservations.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Copyright Â© Magento, Inc. All rights reserved.
```
Comment (robertorubino) [inline InventoryReservationCli/Model/SalableQuantityInconsistency/LoadExistingReservations.php]: I can see a wrong character before the copyright simbol, please compared with the original commit from. MDVA-37291 https://github.com/magento-sparta/inventory/pull/27/files#diff-b22acb9071fb69724ec574b40c0d8c89d92ac047b57428b5d7172674f151d1d3R3

File: InventoryReservationCli/Model/SalableQuantityInconsistency/AddExpectedReservations.php [hunk 2]
```diff
@@ -102,4 +102,20 @@ private function calculateReservationQty(array $data): float
...
+     */
+    private function calculateReservationQty(array $data): float
+    {
+        $qty = $data['qty_ordered'];
+        $qty -= $data['qty_canceled'];
+        $qty -= $data['qty_refunded'];
+        $qty -= $data['is_virtual'] ? $data['qty_invoiced'] : $data['qty_shipped'];
+
+        return (float)$qty;
+    }
```
Comment (robertorubino) [inline InventoryReservationCli/Model/SalableQuantityInconsistency/AddExpectedReservations.php]: This code is not part of ACP2E-294 or MC-42066, but of MC-41222 that has been backported in MDVA-37291 (see https://github.com/magento-sparta/inventory/pull/27/files#diff-55972798e3b836abd2d2297cadd76ae53ead9e2a0a319647b0363f8f373ccdc9R89). As a result, if you check in https://github.com/magento-sparta/inventory/pull/158/files#diff-55972798e3b836abd2d2297cadd76ae53ead9e2a0a319647b0363f8f373ccdc9R89 you can see that in your PR you have 2 methods with the same name "calculateReservationQty"

**Review by robertorubino — CHANGES_REQUESTED** [review]

File: InventoryReservationCli/Model/SalableQuantityInconsistency/LoadExistingReservations.php (hunk 1 above)
Comment (saddane) [reply InventoryReservationCli/Model/SalableQuantityInconsistency/LoadExistingReservations.php]: @robertorubino Fixed. Thanks for catching this

File: InventoryReservationCli/Model/SalableQuantityInconsistency/AddExpectedReservations.php (hunk 2 above)
Comment (saddane) [reply InventoryReservationCli/Model/SalableQuantityInconsistency/AddExpectedReservations.php]: @robertorubino Fixed. Thanks for catching this

**Review by robertorubino — APPROVED** [review]

**Pull request closed without merging**